### PR TITLE
Quartz + Scheduler: Allow to use config in Identity field

### LINKED ISSUE #14967: Quartz: Allow to use a configuration property in identity
**Description**
Most of the fields in the `@Scheduled` annotation allow to use a config property:

```
@Scheduled(cron = "{myMethod.cron.expr}")
```

However, the identity field can't do:

```
@Scheduled(identity = "{job.id}")
```

### PR DESCRIPTION
**Pull request by Sgitario**
These changes allow to use config properties in the `identity` field:

```
@Scheduled(identity = "{job.id}")
```

Fix #14967

### CODE REVIEW

File: extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/PausedSchedulerTest.java [hunk 1]
```diff
@@ -33,15 +34,19 @@
...
     static class Jobs {
+        static final AtomicBoolean IS_WATCHING = new AtomicBoolean(false);
         static final CountDownLatch LATCH = new CountDownLatch(2);
 
         @Scheduled(every = "1s")
         void countDownSecond() {
-            LATCH.countDown();
+            if (IS_WATCHING.get()) {
+                LATCH.countDown();
+            }
```
Comment (Sgitario) [inline extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/PausedSchedulerTest.java:49]: This change is unrelated to my changes, but it was randomly failing for me. Using the is watching boolean, made the test more consistent.

**Comment by Sgitario** [comment]
@machi1990 These changes allow to use confg values in the identity field which is important for Quartz. 
However, I noticed that the test cases are almost duplicated in the scheduler and quartz extensions. To continue doing the same, I added the same test cases in both too, but let me know if this is not correct.

File: extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/PausedSchedulerTest.java (hunk 1 above)
Comment (machi1990) [reply extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/PausedSchedulerTest.java:49]: I've never seen this error before. Do you mind putting this in a separate commit / PR?

File: extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java [hunk 2]
```diff
@@ -206,6 +199,17 @@ public static Duration parseDuration(Scheduled scheduled, String value, String m
         }
     }
 
+    // Keep it public so that we can reuse the logic in the quartz extension
```
Comment (machi1990) [inline extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java:202]: I think we can remove the comment

**Review by machi1990 — COMMENTED** [review]
@Sgitario thanks for the quick PR. 

LGTM, I've added some comments. Can you have a look?

File: extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/PausedSchedulerTest.java (hunk 1 above)
Comment (Sgitario) [reply extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/PausedSchedulerTest.java]: I've deleted my changes here and let's see if the CI complains. If so, I will create a separate PR for this. 
Thanks!

File: extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java (hunk 2 above)
Comment (Sgitario) [reply extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java]: Removed

**Review by machi1990 — APPROVED** [review]
LGTM. 
@mkouba can you have a look too?

File: extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java [hunk 3]
```diff
@@ -154,7 +154,7 @@ public boolean isRunning() {
     }
 
     SimpleTrigger createTrigger(String invokerClass, CronParser parser, Scheduled scheduled, int nameSequence, Config config) {
```
Comment (mkouba) [inline extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java:156]: If we move the logic to the static method, we don't need to inject `Config` at all...

File: extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java [hunk 4]
```diff
@@ -188,13 +185,8 @@ SimpleTrigger createTrigger(String invokerClass, CronParser parser, Scheduled sc
         }
     }
 
-    // Keep it public so that we can reuse the logic in the quartz extension
```
Comment (mkouba) [inline extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java:191]: Maybe we should extract these two static methods to a dedicated class? I'm not sure about the name though...

File: extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java (hunk 3 above)
Comment (Sgitario) [reply extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java]: Thanks for spotting! I've also removed the unused config from QuartzScheduler.

File: extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java (hunk 4 above)
Comment (Sgitario) [reply extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java:191]: Following the same approach in other extensions, I've created a `SchedulerUtils` and moved these two methods and also some static methods that were part of the interface `SchedulerContext` and were also related to configuration.

I left these methods as static but we could make them instance methods.

Let me know if it looks better now.

Comment (mkouba) [reply extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java:191]: The current version looks perfect! Thanks.

**Review by mkouba — APPROVED** [review]

**Pull request merged**